pandas/tests/series/methods/test_between.py

```diff
@@ -38,3 +39,52 @@ def test_between_period_values(self):
         result = ser.between(left, right)
         expected = (ser >= left) & (ser <= right)
         tm.assert_series_equal(result, expected)
+
+    def test_between_inclusive_is_boolean_string(self):  # :issue:`40628`
+        series = Series(date_range("1/1/2000", periods=10))
+        left, right = series[[2, 7]]
+
+        result = series.between(left, right, inclusive="both")
+        expected = (series >= left) & (series <= right)
+        tm.assert_series_equal(result, expected)
+
+        result = series.between(left, right, inclusive="left")
+        expected = (series >= left) & (series < right)
+        tm.assert_series_equal(result, expected)
+
+        result = series.between(left, right, inclusive="right")
+        expected = (series > left) & (series <= right)
+        tm.assert_series_equal(result, expected)
+
+        result = series.between(left, right, inclusive="neither")
+        expected = (series > left) & (series < right)
+        tm.assert_series_equal(result, expected)
+
+        result = series.between(left, right, inclusive=True)
+        expected = (series >= left) & (series <= right)
+        tm.assert_series_equal(result, expected)
+
+        result = series.between(left, right, inclusive=False)
+        expected = (series > left) & (series < right)
+        tm.assert_series_equal(result, expected)
```

Comment:
```suggestion
```

should remove these since they are being tested below explicitly for deprecation warning

Reply:
actually maybe put them in the test `test_between_inclusive_warning` 

The reason is that when someone actually removes these possibilities on a deprecation cycle this test is unaffected, whereas `test_between_inclusive _warning` can simply be deleted.